src/app/app-routing.module.ts

```diff
@@ -1,7 +1,18 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import {AppComponent} from "./app.component";
+import {HomeContainerComponent} from "./hero/home/home-container/home-container.component";
+import {PopularContainerComponent} from "./hero/popular/popular-container/popular-container.component";
+import {PageNotFoundComponent} from "./user/page-not-found/page-not-found.component";
+import {WelcomeComponent} from "./user/welcome/welcome.component";
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path:"", redirectTo:"welcome",component:WelcomeComponent},
+  {path:"welcome",component:WelcomeComponent},
+  {path:"home",component:HomeContainerComponent},
+  {path:"popular",component:PopularContainerComponent},
```

Comment:
you will also need a detail page container to show the details page and a route for it